frontend/src/lib/components/UserLogin.svelte

```diff
@@ -1,30 +1,27 @@
+<svelte:options runes={true} />
 <script lang="ts">
 import { base } from "$app/paths";
 
-import UserLoginUtil from "$lib/components//UserLoginUtil.svelte";
 import AlertMessage from "$lib/components/AlertMessage.svelte";
 
 import { goto } from "$app/navigation";
 import { authCookieLogin, usersCurrentUser } from "$lib/client/services.gen";
 import { type AuthCookieLoginData, type UserRead } from "$lib/client/types.gen";
-import { currentUser } from "$lib/stores/userStore";
+import { user } from "$lib/stores/userStore.svelte";
 import { preventDefault } from "$lib/util";
 import { Button, Card, Heading, Input, Label } from "flowbite-svelte";
 import { _ } from "svelte-i18n";
 
-// TODO: make the remember thing functional again
-// FIXME: make use of the logic of the AdminUser structure and get rid of the UserStore
 async function refresh(): Promise<string> {
 	const returned = await usersCurrentUser();
 	if (returned.error) {
+		user.data = null;
 		console.log("Error getting current user: ", returned.error.detail);
-		currentUser.set(null);
 		return returned.error.detail;
-	} else {
-		console.log("Successfully retrieved active user");
-		currentUser.set(returned.data as UserRead);
-		return "success";
 	}
+	console.log("Successfully retrieved active user");
+	user.data = returned.data as UserRead;
```

Comment:
if you also check that user.data is not undefined above, I don't think you need the as UserRead cast here